apps/admin/backend/src/store.ts

```diff
@@ -850,6 +913,105 @@ export class Store {
     });
   }
 
+  private getTabulationFilterAsSql(
+    electionId: Id,
+    filter: Tabulation.Filter
+  ): [whereParts: string[], params: Bindable[]] {
+    const whereParts = ['cvrs.election_id = ?'];
+    const params: Bindable[] = [electionId];
+
+    if (filter.ballotStyleIds) {
+      whereParts.push(
+        `cvrs.ballot_style_id in ${asQueryPlaceholders(filter.ballotStyleIds)}`
+      );
+      params.push(...filter.ballotStyleIds);
+    }
+
+    if (filter.precinctIds) {
+      whereParts.push(
+        `cvrs.precinct_id in ${asQueryPlaceholders(filter.precinctIds)}`
+      );
+      params.push(...filter.precinctIds);
+    }
+
+    if (filter.votingMethods) {
+      whereParts.push(
+        `cvrs.ballot_type in ${asQueryPlaceholders(filter.votingMethods)}`
+      );
+      params.push(...filter.votingMethods);
+    }
+
+    if (filter.batchIds) {
+      whereParts.push(
+        `cvrs.ballot_type in ${asQueryPlaceholders(filter.batchIds)}`
+      );
+      params.push(...filter.batchIds);
+    }
+
+    if (filter.precinctIds) {
+      whereParts.push(
+        `cvrs.ballot_type in ${asQueryPlaceholders(filter.precinctIds)}`
+      );
+      params.push(...filter.precinctIds);
+    }
+
+    return [whereParts, params];
+  }
+
+  /**
+   * Returns an iterator of cast vote records for tabulation purposes. Filters
+   * the cast vote records by specified filters.
+   */
+  *getCastVoteRecords({
+    electionId,
+    election,
+    filter,
+  }: {
+    electionId: Id;
+    election: Election;
+    filter: Tabulation.Filter;
+  }): Generator<Tabulation.CastVoteRecord> {
+    const [whereParts, params] = this.getTabulationFilterAsSql(
+      electionId,
+      replacePartyIdFilter(filter, election)
```

Comment:
We have to convert the filters available to the API into a filter processable by the store. At this time, that only means replacing filters on `partyId` with filters on `ballotStyleId`, but there could be more in the future.

In the `partyId` case, I wonder if it's worth just adding `partyId` to the `cvrs` table and the `Tabulation.CastVoteRecord` object to avoid the special casing around party that you see here and in tabulation code.

Reply:
I am curious whether we'll have more cases like `partyId` (a.k.a. filters and groupings based on combinations of other more fundamental filters and groupings). It might be one among many, and we don't want to special case it in the data model but just have good patterns of handling it. Or, it might be unique - in which case I'd like to circle back and clean it up later.